ext/geos_c_impl/factory.c

```diff
@@ -929,7 +848,8 @@ VALUE rgeo_geos_klasses_and_factories_eql(VALUE obj1, VALUE obj2)
   }
   else {
     factory = RGEO_GEOMETRY_DATA_PTR(obj1)->factory;
-    result = rb_funcall(factory, RGEO_FACTORY_DATA_PTR(factory)->globals->id_eql, 1, RGEO_GEOMETRY_DATA_PTR(obj2)->factory);
+    /* No need to cache the internal here (https://ips.fastruby.io/4x) */
+    result = rb_funcall(factory, rb_intern("eql?"), 1, RGEO_GEOMETRY_DATA_PTR(obj2)->factory);
```

Comment:
We extended the usage of this benchmark to `"hash"` and `"cast"` as well, the idea being that their usage does not call for such precise performance optimisation (otherwise, we should not be calling `rb_funcall` either).

Performance matters, however we're working with geometries that are much computation heavier than featching a value in a global symbol cache, hence we decided to pick our fights and prefer readability here.

The benchmark was made with:

```ruby
require "benchmark/ips"
load "#{__dir__}/lib/rgeo.rb"

# Puts a Centered, bold and inverted text for better visibility.
def shout_out(string)
  puts "\n\e[7;1m#{string.center(75)}\e[27;0m\n\n"
end

shout_out("SMALL POLYGON")

def ring(pts, factory: RGeo::Geos.factory)
  factory.linear_ring(
    pts.map { factory.point(_1, _2) }
  )
end

def polygon(pts, factory: RGeo::Geos.factory)
  pts = pts.each_slice(2) if pts.first.is_a?(Numeric) && pts.size.even?
  factory.polygon(
    ring(pts, factory: factory)
  )
end

poly = polygon([0, 0,    0, 1,    1, 1,    1, 0,    0, 0])

Benchmark.ips do |x|
  x.config(time: 20)
  x.report("rb_intern") { poly == poly }
  x.report("cached") { poly == poly }
  x.hold!("temp")
  x.compare!
end
```

and commenting the polygon equality part to only have this class used in c.